crates/ruff_linter/src/rules/pyflakes/snapshots/ruff_linter__rules__pyflakes__tests__F401_F401_24____init__.py.snap

```diff
@@ -0,0 +1,42 @@
+---
+source: crates/ruff_linter/src/rules/pyflakes/mod.rs
+---
+__init__.py:23:8: F401 [*] `sys` imported but unused; consider removing, adding to `__all__`, or using a redundant alias
+   |
+23 | import sys # F401: remove unused
+   |        ^^^ F401
+   |
+   = help: Remove unused import: `sys`
+
+ℹ Unsafe fix
+20 20 | 
+21 21 | 
+22 22 | 
+23    |-import sys # F401: remove unused
+24 23 | 
+25 24 | 
+26 25 | 
+
+__init__.py:43:15: F401 [*] `.unused` imported but unused; consider removing, adding to `__all__`, or using a redundant alias
+   |
+43 | from . import unused # F401: change to redundant alias
+   |               ^^^^^^ F401
+   |
+   = help: Remove unused import: `.unused`
```

Comment:
This help message is wrong, it should say something about exporting.

Reply:
Yeah, I didn't touch the diagnostic generation code. My bad. I'll take a look at improving that.

Reply:
Usually you have the violation take an enum then use that to generate the proper message

Reply:
e.g. https://github.com/astral-sh/ruff/blob/a6d892b1f4b45bfed918f3240218d5fcd39d3099/crates/ruff_linter/src/rules/flake8_bandit/rules/bad_file_permissions.rs#L43-L52

Reply:
Thanks! I hadn't picked up on this pattern.

Reply:
I changed it so that when in __init__.py and the import is first-party and there's no __all__ it says to "use a redundant alias" and if there is a __all__ it says "add to __all__"; otherwise it falls back to the text "remove unused import". This isn't perfect, because it means in the `import foo as bar` case it recommends one of the export texts, not "remove unused import".